src/environmentd/src/bin/environmentd/main.rs

```diff
@@ -717,6 +726,18 @@ fn run(mut args: Args) -> Result<(), anyhow::Error> {
     );
     let persist_clients = Arc::new(persist_clients);
     let orchestrator = Arc::new(TracingOrchestrator::new(orchestrator, args.tracing.clone()));
+
+    if let Some(instance_directory) = &args.instance_directory {
+        if instance_directory.exists() && !instance_directory.is_dir() {
+            bail!("`--instance-directory` must refer to a directory");
+        }
+
+        if !instance_directory.exists() {
+            std::fs::create_dir_all(instance_directory)
+                .context("failed to create `--instance-directory`")?;
+        }
+    }
```

Comment:
This validation should happen inside of environmentd/lib.rs, not here, since it's not related to the CLI arguments specifically.

Reply:
Wait, actually, I don't think envd should be creating the scratch directory at all! It should be the individual clusters.